pkg/storage/kubernetes/client.go

```diff
@@ -135,3 +137,18 @@ func (i *Client) DeleteOverlappingIP(ctx context.Context, clusterWideIP *whereab
 	return i.client.WhereaboutsV1alpha1().OverlappingRangeIPReservations(clusterWideIP.GetNamespace()).Delete(
 		ctxWithTimeout, clusterWideIP.GetName(), metav1.DeleteOptions{})
 }
+
+func (i *Client) PatchOverlappingIP(ctx context.Context, clusterWideIP *whereaboutsv1alpha1.OverlappingRangeIPReservation) (*whereaboutsv1alpha1.OverlappingRangeIPReservation, error) {
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, storage.RequestTimeout)
+	defer cancel()
+
+	patchBytes, err := json.Marshal(map[string]interface{}{
+		"spec": clusterWideIP.Spec,
+	})
+	if err != nil {
+		return nil, logging.Errorf("Error marshaling overlappingrangeipreservations.whereabouts.cni.cncf.io spec: %v", err)
+	}
+
+	return i.client.WhereaboutsV1alpha1().OverlappingRangeIPReservations(clusterWideIP.GetNamespace()).Patch(
```

Comment:
question: does this have to be Patch, don't we have an Update() for that?  (I might be completely wrong)

Reply:
be it as it may, isn't https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict needed?

Reply:
The IP reconciler runs on multiple nodes, there will be racing conditions if using Update() in this case. However, using Patch() could cause too many API requests. We may need a leader election here.

I agree that we need to add a retry mechanism. This is a general problem in the IP reconciler, it doesn’t retry when reconciliation fails.